komodo-spi/src/main/java/org/komodo/spi/utils/KeyInValueHashMap.java

```diff
@@ -21,22 +21,18 @@
  */
 package org.komodo.spi.utils;
 
-import java.util.AbstractMap;
 import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.LinkedHashMap;
 
 /**
  * A {@link HashMap} which stores values that also reference their
- * own keys. More memory efficient than storing separate references
- * to the keys and values.
+ * own keys.
  *
  * @param <K> key
  * @param <V> value
  */
 @SuppressWarnings( "unchecked" )
-public class KeyInValueHashMap<K, V> extends AbstractMap<K, V> {
+public class KeyInValueHashMap<K, V> extends LinkedHashMap<K, V> {
```

Comment:
> This implementation was meant to improve memory footprint.

Which it did not.  A HashSet simply wraps a HashMap, then the usage of a EntryWrapper adds another Object/reference into the picture.  If you profile the retained heap size of this vs. the old implementation you will see that the old was larger.

> By making it a sorted map, it raises the memory profile for an implementation that only occasionally requires the keys to be sorted.

Linked does not imply sorted.  Linked only maintains the insertion order, not the natural ordering.  You can certainly change that to just a HashMap, as LinkedHashMap does add a minor amount of overhead to each entry (2 references).  However what I've seen a number of times on the Teiid side are unit tests or other places in the code that inadvertently rely on the ordering of the hashing, only to have that change in the next java version.

Reply:
You seem to have confused my comments. I am not suggesting that the 'new' version is larger/smaller than the 'old' version. The memory footprint comment refers to specifically using a LinkedHashMap which, as you say, includes additional overhead, compared to an AbstractMap.

'sorted' was obviously the incorrect word.

Again, if you have encountered instances where the solution is better, please say so in the commit message. Such comments preclude questioning and misunderstanding.

Reply:
Yes, the confusion stemmed from the usage of sorted.

> Such comments preclude questioning and misunderstanding.

I'm honestly not trying to shirk comments.  Since the web.xml and this appeared to be minor changes upon initial review I simply included in this pull request - I will separate such commits for clarity.

Reply:
Thanks very much. It is much appreciated.